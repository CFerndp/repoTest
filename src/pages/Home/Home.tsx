import React from "react";
import PageTemplate from "../../components/PageTemplate/PageTemplate.tsx";
import {Link} from "react-router-dom";
import Button from "../../components/Button/Button.tsx";
import {PATHS} from "../../routes/paths.ts";
import Title from "../../components/Title/Title.tsx";

const Home: React.FC = () => (
    <PageTemplate>
        <Title>
            ¡Hola! Bienvenido a tu gestor de tareas.
        </Title>
        <p className="mt-2 w-50">
            Esperamos poder ayudarte, por favor, si te has registrado, pincha en el botón de abajo para comenzar:
        </p>
        <Link to={PATHS.TASK}><Button label="Comenzar" /></Link>
    </PageTemplate>
)

export default Home
