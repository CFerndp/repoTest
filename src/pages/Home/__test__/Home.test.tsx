import {describe, expect, test} from "vitest";
import {render, screen} from "@testing-library/react";
import Home from "../Home.tsx";
import {BrowserRouter} from "react-router-dom";
import userEvent from "@testing-library/user-event";


describe('Given Home Page component', ()=>{
    describe('When it is setted properly', ()=>{
        test('Then it is ok rendered',async  ()=>{
            const component = render(<Home />, {wrapper: BrowserRouter});
            expect(component).toMatchSnapshot()
        })
    })

    describe('When the user clicks on next button', ()=>{
        test('Then the user goes to the next page',async  ()=>{
             render(<Home />, {wrapper: BrowserRouter});
             const user = userEvent.setup()

            await user.click(screen.getByTestId('button-next-page'))

        })
    })
})
