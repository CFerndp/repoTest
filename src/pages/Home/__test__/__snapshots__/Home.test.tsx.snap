// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given Home Page component > When it is setted properly > Then it is ok rendered 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="border-b-2 border-blue-500 py-3 flex flex-row justify-between"
      >
        <a
          href="/"
        >
          <h1
            class="text-3xl font-bold ml-5"
          >
            Mi Task Controller
          </h1>
        </a>
        <button
          class="p-2 border-2 bg-blue-500 border-white rounded-lg text-white p-2 border-2 bg-blue-500 border-white rounded-lg text-white mr-5"
          data-testid="button"
        >
          Registrarse
        </button>
      </div>
      <div
        class="container m-6"
      >
        <h1
          class="text-2xl font-bold underline"
        >
          ¡Hola! Bienvenido a tu gestor de tareas.
        </h1>
        <p
          class="mt-2 w-50"
        >
          Esperamos poder ayudarte, por favor, si te has registrado, pincha en el botón de abajo para comenzar:
        </p>
        <a
          href="/tasks"
        >
          <button
            class="p-2 border-2 bg-blue-500 border-white rounded-lg text-white undefined"
            data-testid="button-next-page"
          >
            Comenzar
          </button>
        </a>
      </div>
      <div
        class="border-t-2 border-blue-500 py-3"
      >
        <h1
          class="text-m font-bold text-center"
        >
          Hecho con amor para que Alex no muera en el intento del Bootcamp :)
        </h1>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="border-b-2 border-blue-500 py-3 flex flex-row justify-between"
    >
      <a
        href="/"
      >
        <h1
          class="text-3xl font-bold ml-5"
        >
          Mi Task Controller
        </h1>
      </a>
      <button
        class="p-2 border-2 bg-blue-500 border-white rounded-lg text-white p-2 border-2 bg-blue-500 border-white rounded-lg text-white mr-5"
        data-testid="button"
      >
        Registrarse
      </button>
    </div>
    <div
      class="container m-6"
    >
      <h1
        class="text-2xl font-bold underline"
      >
        ¡Hola! Bienvenido a tu gestor de tareas.
      </h1>
      <p
        class="mt-2 w-50"
      >
        Esperamos poder ayudarte, por favor, si te has registrado, pincha en el botón de abajo para comenzar:
      </p>
      <a
        href="/tasks"
      >
        <button
          class="p-2 border-2 bg-blue-500 border-white rounded-lg text-white undefined"
          data-testid="button-next-page"
        >
          Comenzar
        </button>
      </a>
    </div>
    <div
      class="border-t-2 border-blue-500 py-3"
    >
      <h1
        class="text-m font-bold text-center"
      >
        Hecho con amor para que Alex no muera en el intento del Bootcamp :)
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
